include/dslash_helper.cuh

```diff
@@ -264,13 +282,25 @@ namespace quda
 
     int pack_threads; // really number of face sites we have to pack
     int_fastdiv blocks_per_dir;
+    int sites_per_block;
     int dim_map[4];
     int active_dims;
     int pack_blocks; // total number of blocks used for packing in the dslash
+    int ext_dims;    // dimension to run in the exterior Dslash
+    int ext_blocks;
+
+    // for shmem ...
+    static constexpr bool packkernel = false;
+    void *packBuffer[4 * QUDA_MAX_DIM];
+    int neighbor_ranks[2 * QUDA_MAX_DIM];
+    int bytes[2 * QUDA_MAX_DIM];
+    volatile long *sync_arr;
```

Comment:
Why is this volatile?  Can we replace with `std::atomic`?

Reply:
Guess we could get this when using placement new, but it might prevent us better `nvshmem_wait_<whatever>` things in the future. I'd defer that to future optimization. But getting this to be atomic would be nicer.